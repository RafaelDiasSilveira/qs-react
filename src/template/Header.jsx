import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
    <NavLink className="navbar-brand" to="/" >
      QS-React
    </NavLink>

    <button
      className="navbar-toggler"
      type="button"
      data-toggle="collapse"
      data-target="#navbarText"
      aria-controls="navbarText"
      aria-expanded="false"
      aria-label="Toggle navigation"
    >
      <span className="navbar-toggler-icon" />
    </button>


    <div className="collapse navbar-collapse" id="navbarText">

      <ul className="navbar-nav mr-auto">
        <li activeClassName="active">
          <NavLink className="nav-link" to="/" exact>Home</NavLink>
        </li>
        <li activeClassName="active">
          <NavLink className="nav-link" to="/users" exact>Usu√°rios</NavLink>
        </li>
        <li activeClassName="active">
          <NavLink className="nav-link" to="/books" exact>Livros</NavLink>
        </li>
      </ul>

    </div>

  </nav>
);

export default Header;
