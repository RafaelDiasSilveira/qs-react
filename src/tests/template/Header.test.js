import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../template/Header';

let wrapper;

beforeAll(() => {
  wrapper = shallow(<Header />);
});

test('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should contain the correct amount of links', () => {
  const result = wrapper.find('NavLink')

  expect(result).toHaveLength(5);
});
