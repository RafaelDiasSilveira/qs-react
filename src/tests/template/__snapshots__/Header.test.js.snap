// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="navbar-brand"
    to="/"
  >
    QS-React
  </NavLink>
  <button
    aria-controls="navbarText"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarText"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarText"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        activeClassName="active"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/"
        >
          Estante
        </NavLink>
      </li>
      <li
        activeClassName="active"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/users"
        >
          Usu√°rios
        </NavLink>
      </li>
      <li
        activeClassName="active"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/books"
        >
          Livros
        </NavLink>
      </li>
      <li
        activeClassName="active"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/profile"
        >
          Meu Perfil
        </NavLink>
      </li>
    </ul>
  </div>
</nav>
`;
